src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -126,12 +127,31 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 );
                 String defaultsFilePath = DefaultUseCases.getDefaultsFileByUseCaseName(useCase);
                 useCaseDefaultsMap = ParseUtils.parseJsonFileToStringToStringMap("/" + defaultsFilePath);
-
-                if (request.hasContent()) {
+                List<String> requiredParams = DefaultUseCases.getRequiredParamsByUseCaseName(useCase);
+
+                if (request.hasContent() == false) {
+                    if (requiredParams.size() != 0) {
```

Comment:
Combine these with an &&.  Also `size != 0` is better `!requiredParams.isEmpty()`.

Reply:
If we combine these, then we will error out if a user attempts to use a use-case that requires no parameters, such as the `local_neural_sparse_search_bi_encoder`